pkg/controller/scylladbmonitoring/sync_prometheus.go

```diff
@@ -0,0 +1,513 @@
+		"scyllaDBMonitoringName": sm.Name,
+	})
+}
+
+func makePrometheus(sm *scyllav1alpha1.ScyllaDBMonitoring) (*monitoringv1.Prometheus, string, error) {
+	spec := getPrometheusSpec(sm)
+
+	var volumeClaimTemplate *monitoringv1.EmbeddedPersistentVolumeClaim
+	if spec != nil && spec.Storage != nil {
+		volumeClaimTemplate = &monitoringv1.EmbeddedPersistentVolumeClaim{
+			EmbeddedObjectMetadata: monitoringv1.EmbeddedObjectMetadata{
+				Name:        fmt.Sprintf("%s-prometheus", sm.Name),
+				Labels:      spec.Storage.VolumeClaimTemplate.Labels,
+				Annotations: spec.Storage.VolumeClaimTemplate.Annotations,
+			},
+			Spec: spec.Storage.VolumeClaimTemplate.Spec,
+		}
+
+	}
+
+	affinity := corev1.Affinity{}
+	var tolerations []corev1.Toleration
+	if spec != nil && spec.Placement != nil {
+		affinity.NodeAffinity = spec.Placement.NodeAffinity
+		affinity.PodAffinity = spec.Placement.PodAffinity
+		affinity.PodAntiAffinity = spec.Placement.PodAntiAffinity
+
+		tolerations = spec.Placement.Tolerations
+	}
+
+	var resources corev1.ResourceRequirements
+	if spec != nil {
+		resources = spec.Resources
+	}
+
+	return prometheusv1assets.PrometheusTemplate.RenderObject(map[string]any{
+		"namespace":              sm.Namespace,
+		"scyllaDBMonitoringName": sm.Name,
+		"volumeClaimTemplate":    volumeClaimTemplate,
+		"affinity":               affinity,
+		"tolerations":            tolerations,
+		"resources":              resources,
+	})
+}
+
+func makePrometheusIngress(sm *scyllav1alpha1.ScyllaDBMonitoring) (*networkingv1.Ingress, string, error) {
+	ingressOptions := getPrometheusIngressOptions(sm)
+	if ingressOptions == nil {
+		return nil, "", nil
+	}
+
+	if ingressOptions.Disabled != nil && *ingressOptions.Disabled == true {
+		return nil, "", nil
+	}
+
+	if len(ingressOptions.DNSDomains) == 0 {
+		return nil, "", nil
+	}
+
+	return prometheusv1assets.PrometheusIngressTemplate.RenderObject(map[string]any{
+		"scyllaDBMonitoringName": sm.Name,
+		"dnsDomains":             ingressOptions.DNSDomains,
+		"ingressAnnotations":     ingressOptions.Annotations,
+		"ingressClassName":       ingressOptions.IngressClassName,
+	})
+}
+
+func (smc *Controller) syncPrometheus(
+	ctx context.Context,
+	sm *scyllav1alpha1.ScyllaDBMonitoring,
+	configMaps map[string]*corev1.ConfigMap,
+	secrets map[string]*corev1.Secret,
+	services map[string]*corev1.Service,
+	serviceAccounts map[string]*corev1.ServiceAccount,
+	roleBindings map[string]*rbacv1.RoleBinding,
+	ingresses map[string]*networkingv1.Ingress,
+	prometheuses map[string]*monitoringv1.Prometheus,
+	prometheusRules map[string]*monitoringv1.PrometheusRule,
+	serviceMonitors map[string]*monitoringv1.ServiceMonitor,
+) ([]metav1.Condition, error) {
+	var progressingConditions []metav1.Condition
+
+	prometheusServingCertChainConfig := &okubecrypto.CertChainConfig{
+		CAConfig: &okubecrypto.CAConfig{
+			MetaConfig: okubecrypto.MetaConfig{
+				Name:   fmt.Sprintf("%s-prometheus-serving-ca", sm.Name),
```

Comment:
I am not in favour of e2e / conformance test using those, it would make it very easy to break compatibility, this way the change has to be explicitly made to the e2e / conformance test itself

Reply:
agree, let's keep it